File: TransCoder_tokenized_test_set_functions/java/MAXIMUM_NUMBER_SEGMENTS_LENGTHS_B_C.java
static int maximumSegments ( int n, int a, int b, int c ) {
  int dp [ ] = new int [ n + 10 ];
  Arrays . fill ( dp, - 1 );
  dp [ 0 ] = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( dp [ i ] != - 1 ) {
      if ( i + a <= n ) dp [ i + a ] = Math . max ( dp [ i ] + 1, dp [ i + a ] );
      if ( i + b <= n ) dp [ i + b ] = Math . max ( dp [ i ] + 1, dp [ i + b ] );
      if ( i + c <= n ) dp [ i + c ] = Math . max ( dp [ i ] + 1, dp [ i + c ] );
    }
  }
  return dp [ n ];
}

File: results/py/MAXIMUM_NUMBER_SEGMENTS_LENGTHS_B_C.cpp
int f_filled ( int n , int a , int b , int c ) {
  int dp [ n + 10 ] ;
  dp [ 0 ] = 0 ;
  for ( int i = 0 ;
  i <= n ;
  i ++ ) {
    if ( ( dp [ i ] != - 1 ) && ( i + a <= n ) ) {
      if ( ( i + a <= n ) && ( i + b <= n ) && ( i + c <= n ) ) {
        dp [ i + a ] = max ( dp [ i ] + 1 , dp [ i + a ] ) ;
      }
      if ( ( i + b <= n ) && ( i + c <= n ) && ( i + d <= n ) ) {
        dp [ i + d ] = max ( dp [ i ] + 1 , dp [ i + d ] ) ;
      }
    }
  }
  return dp [ n ] ;
}
